File: OnlinePharmacyAndPrescriptionSystem/Models/Feedback.cs
using System.ComponentModel.DataAnnotations;

namespace OnlinePharmacyAndPrescriptionSystem.Models
{
    public class Feedback
    {
        [Key]
        public int Id { get; set; }

        public string UserName { get; set; }

        public int MedicineId { get; set; }

        public int Rating { get; set; }

        public string Comment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: src/app/features/customer/feedback/feedback.ts
import { Component, signal } from '@angular/core';
import { FeedbackService } from '../../../core/services/feedback-service';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-feedback',
  imports: [FormsModule,CommonModule],
  templateUrl: './feedback.html',
  styleUrl: './feedback.css',
})
export class Feedback {

  model = {
    userName: 'devira',
    medicineId: 1,
    rating: 5,
    comment: ''
  };

  message = signal('');

  constructor(private service: FeedbackService) {}

  submit() {
    this.service.add(this.model).subscribe(() => {
      this.message.set('Feedback Submitted Successfully');
      this.model.comment = '';
    });
  }
}
